### PR TITLE
Nodejs/week3/warmup

### PR DESCRIPTION
**Pull request by Gohar33**

### CODE REVIEW

File: nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js [hunk 1]
```diff
@@ -10,11 +10,9 @@ router.get("/", async (request, response) => {
```
Comment (cecastosic) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js:10]: this shouldn't be part of the PR since it is homework for week 2

File: nodejs/week3/warmup/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+app.get('/calculator/substract', function (req, res) {
+    const values = Object.values(req.query).flat()
+    const subtsract = values.reduce((acc, val) => Number(acc - Number(val)))
+    res.send(` ${subtsract}`)
+});
+
+app.get('/calculator/add', function (req, res) {
+    const values = Object.values(req.query).flat()
+    const subtsract = values.reduce((acc, val) => (Number(acc) + Number(val)))
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:25]: const subtract = ...

File: nodejs/week3/warmup/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+app.get('/calculator/multiply', function (req, res) {
+    const values = Object.values(req.query).flat()
+    const multiply = values.reduce((acc, val) => (acc) * (val))
+    res.send(` ${multiply}`)
+});
+
+app.get('/calculator/substract', function (req, res) {
+    const values = Object.values(req.query).flat()
+    const subtsract = values.reduce((acc, val) => Number(acc - Number(val)))
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:19]: const subtract = ...

File: nodejs/week3/warmup/index.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+app.get("/", (req, res) => res.send("nodejs week3 homework"));
+
+app.get('/calculator/multiply', function (req, res) {
+    const values = Object.values(req.query).flat()
+    const multiply = values.reduce((acc, val) => (acc) * (val))
+    res.send(` ${multiply}`)
+});
+
+app.get('/calculator/substract', function (req, res) {
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:17]: app.get('/calculator/subtract'

**Review by cecastosic — COMMENTED** [review]
Good job 💪 
- Missing POST requests for addition, subtraction and division
- What should happen if the query parameters are not numbers? For example /calculator/multiply?firstParam=a&secondParam=b
- There are also parts of the code that are repeating and could be optimized

**Comment by Gohar33** [comment]
Thanks a lot for your feedback Svetlana <3  I'm very ashamed of my knowledge of  English,  until now I know that word as substract :(

**Comment by Gohar33** [comment]
Since this pull request was merged already, I implemented the feedback and committed all changes directly in master branch
